pkg/apis/autoscaling/types.go

```diff
@@ -303,3 +303,106 @@ type HorizontalPodAutoscalerList struct {
+// type ClusterAutoscalerSpec struct {
+// }
+
+// Type of ClusterAutoscalerCondition
+type ClusterAutoscalerConditionType string
+
+const (
+	// Condition that explains what is the current health of ClusterAutoscaler or its node groups.
+	ClusterAutoscalerHealth ClusterAutoscalerConditionType = "Health"
+	// Condition that explains what is the current status of a node group with regard to
+	// scale down activities.
+	ClusterAutoscalerScaleDown ClusterAutoscalerConditionType = "ScaleDown"
+	// Condition that explains what is the current status of a node group with regard to
+	// scale down activities.
+	ClusterAutoscalerScaleUp ClusterAutoscalerConditionType = "ScaleUp"
+)
+
+// Status of ClusterAutoscalerCondition.
+type ClusterAutoscalerConditionStatus string
+
+const (
+	// Statuses for Health condition type.
+	ClusterAutoscalerHealthy   ClusterAutoscalerConditionStatus = "Healthy"
+	ClusterAutoscalerUnhealthy ClusterAutoscalerConditionStatus = "Unhealthy"
+
+	// Statuses for ScaleDown condition type.
+	ClusterAutoscalerCandidatesPresent ClusterAutoscalerConditionStatus = "CandidatesPresent"
+	ClusterAutoscalerNoCandidates      ClusterAutoscalerConditionStatus = "NoCandidates"
+
+	// Statuses for ScaleUp condition type.
+	ClusterAutoscalerCandidatesNeeded ClusterAutoscalerConditionStatus = "Needed"
+	ClusterAutoscalerNotNeeded        ClusterAutoscalerConditionStatus = "NotNeeded"
+	ClusterAutoscalerInProgress       ClusterAutoscalerConditionStatus = "InProgress"
+	ClusterAutoscalerNoActivity       ClusterAutoscalerConditionStatus = "NoActivity"
+)
+
+// ClusterAutoscalerCondition describes some aspect of ClusterAutoscaler work.
+type ClusterAutoscalerCondition struct {
+	// Defines the aspect that the condition describes. For example, it can be Health or ScaleUp/Down activity.
+	Type ClusterAutoscalerConditionType
+	// Status of the condition. Tells how given aspect
+	Status ClusterAutoscalerConditionStatus
+	// Free text extra information about the condition. It may contain some
+	// extra debugging data, like why the cluster is unhealthy.
+	// +optional
+	Message string
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	// +optional
+	Reason string
+	// Last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time
+	// Since when the condition was in the given state.
+	// +optional
+	LastTransitionTime metav1.Time
+}
+
+// Status of ClusterAutoscaler
+type ClusterAutoscalerStatus struct {
+	// Status information of individual node groups on which CA works.
+	// +optional
+	NodeGroupStatuses []NodeGroupStatus
+	// Conditions that apply to the whole autoscaler.
+	// +optional
+	ClusterwideConditions []ClusterAutoscalerCondition
+}
+
+// Status of a group of nodes controlled by ClusterAutoscaler.
+type NodeGroupStatus struct {
+	// Name of the node group. On GCE it will be equal to MIG url, on AWS it will be ASG name, etc.
+	ProviderID string
```

Comment:
Maybe rename to `NodeGroupProviderID`? Currently it looks like ID of provider (not node gropu).